### PR TITLE
feet: implement endpoint to fetch billing plan details

### PR DESCRIPTION
**Pull request by FreemanBoss**
This pull request introduces a new endpoint `GET /api/v1/organizations/{organization_id}/billing-plans/{plan_name}` that allows user to retrieve a billing plans associated with a specified organization. The endpoint responds with a JSON object containing the status of the request, a message, and an array of billing plans, each with its ID, name, price, and features.

```json
{
  "status": 200,
  "message": "details fetched successfully",
  "data": {
    "plan": {
      "id": "premium",
      "name": "Premium Plan",
      "price": 20.00,
      "features": ["Feature 1", "Feature 2", "Feature 3"],
      "description": "Detailed description of the Premium Plan."
    }
  }
}
```
Related Issue (https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issue/298)
#298 

## Motivation and Context

This change is required to provide organizations with the ability to view available billing plans. It addresses the need for enhanced visibility of billing options, allowing organizations to make informed decisions about subscription levels and features they may want to use.

## How Has This Been Tested?

The new endpoint has been tested using Postman and swagger to ensure that the correct billing plans are fetched based on valid organization IDs. Testing included the following:

 Sending requests with valid authorization tokens.
 Verifying the response format, status code, and data integrity.
 Ensuring that appropriate status codes are returned for unauthorized tokens

Types of changes
Screenshot_2024-07-25-19-37-33-963_com android chrome

 Bug fix (non-breaking change which fixes an issue)
 New feature (non-breaking change which adds functionality)
 Breaking change (fix or feature that would cause existing functionality to change)
Checklist:

 - My code follows the code style of this project.
 - My change requires a change to the documentation.
-  I have updated the documentation accordingly.
 - I have read the CONTRIBUTING document.
- I have added tests to cover my changes.
- All new and existing tests passed.

### CODE REVIEW

File: tests/v1/test_plan.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import pytest
```
Comment (joboy-dev) [inline tests/v1/test_plan.py:1]: Write tests to mock the database
Check the current test files for how to go about it

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make requested changes

**Comment by SundayMba** [comment]
Attach POSTMAN image of the test.

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

File: tests/v1/test_plan.py (hunk 1 above)
Comment (FreemanBoss) [reply tests/v1/test_plan.py]: @joboy-dev done.

**Comment by Goketech** [comment]
fix conflicts

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Resolve merge conflicts

**Comment by FreemanBoss** [comment]
@joboy-dev please check again 
It's someone that actually broke my test

**Comment by FreemanBoss** [comment]
@Goketech you can check now

**Pull request closed without merging**